### DISCUSSION TITLE
Tokens using W3C Design Token spec format

### DISCUSSION THREAD

**Discussion by adbutterfield** [chakra-ui/panda#2366] [Ideas]
I'm implementing a new component library for the new design system where I work. On the design side, we're going to start using [Figma variables](https://help.figma.com/hc/en-us/articles/15339657135383-Guide-to-variables-in-Figma). A nice feature of using variables, is that you can export them in [WC3 Design Token spec format](https://second-editors-draft.tr.designtokens.org/format/), like with a GitHub Actions workflow (there's a video tutorial on how to do that in that Figma link above).

I'm going to write some transformer code to use the variables with Panda. But it would be nice to be able to consume the variables directly. It's a cool feature that Panda could have that the other css libraries don't!

I know Figma variables are still in beta and don't have everything yet. And the Design Token spec is also not finalized. But something to think about!

**Comment by adbutterfield** [comment]
I totally forgot that it says in the docs, Panda token format is inspired by the W3C format...

**Comment by coopbri** [comment]
Agree! Made a similar discussion here: https://github.com/chakra-ui/panda/discussions/3191

**Comment by coopbri** [comment]
@adbutterfield is the transformer code you wrote public? Maybe we could collaborate on it.

**Reply by adbutterfield** [reply]
Currently no. I came up with a very less than ideal workflow at the moment, that uses a custom Figma plugin to export the variables as a big JSON object. Then some scripts in the repo to take that JSON file and transform it.

I think an ideal workflow would be to use [tokens studio](https://tokens.studio/plugin), with a github actions workflow to process the tokens and create a PR, whenever tokens are updated.
